File: credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long number = get_long("Number: ");

    int digits = 0;
    int sum = 0;
    int first;
    int second;
    bool everyOther = true;
    while (number > 0)
    {
        if (number < 100 && number > 9)
        {
            first = number / 10;
            second = number % 10;
        }
        if (everyOther)
        {
            sum += number % 10;
        }
        else
        {
            int timesTwo = (number % 10) * 2;
            sum += timesTwo % 10 + timesTwo / 10;
        }
        everyOther = !everyOther;
        digits++;
        number = number / 10;
    }
    if (sum % 10 == 0)
    {
        if (digits == 15 && first == 3 && (second == 4 || second == 7))
        {
            printf("AMEX\n");
        }
        else if (digits == 16 && first==5 && second>0 && second<6)
        {
            printf("MASTERCARD\n");
        }
        else if((digits==13 || digits==16) && first==4)
        {
            printf("VISA\n");
        }
        else
        {
            printf("INVALID\n");
        }
    }
    else
    {
        printf("INVALID\n");
    }
}

File: Week6/sentimental-credit/credit.py
from cs50 import get_int

credit = get_int("Number: ")
parity = False
sum = 0
count = 0
while credit != 0:
    if credit > 9 and credit < 100:
        firstTwo = credit

    if parity == False:
        sum += credit % 10
    else:
        double = 2 * (credit % 10)
        sum += double // 10 + double % 10
    credit = credit // 10
    count += 1
    parity = not parity

if sum % 10 == 0:
    good = True
else:
    good = False

if good:
    if count == 15 and (firstTwo == 34 or firstTwo == 37):
        print("AMEX\n")
    elif count == 16 and firstTwo // 10 == 5 and firstTwo % 10 > 0 and firstTwo % 10 < 6:
        print("MASTERCARD\n")
    elif (count == 13 or count == 16) and firstTwo // 10 == 4:
        print("VISA\n")
    else:
        print("INVALID\n")
else:
    print("INVALID\n")